docs/DeveloperGuide.md

```diff
@@ -244,8 +244,47 @@ _{more aspects and alternatives to be added}_
 
 #### Proposed Implementation
 
+The proposed implementation for Remark is similar to that of other details of Person.
+Firstly we include a Remark class under `seedu/address/Person/Model`.
+This class will contain all details about a remark, namely its Regex requirements and
+the string which is the remark itself.
+
+Note that a Remark field is optional when adding a Person entry into the AddressBook.
+Hence, the default value for a Remark is an empty String.
+
+There are only 2 commands which can modify the Remark for a Person. These are the 
+`add` command and the `edit` command.
+
+1. `add` Command (upon instantiating)
+   1. If the remark field is specified when instantiating a Person, for example:
+      `add n=JohnDoe p=98765432 e=JohnDoe@gmail.com a=Clementi r=REMARK`, their remark
+      will be REMARK. 
+   2. If not specified, it will be an empty String.
+2. `edit` Command (after instantiating)
+   1. If the remark field is specified when editing a Person, for example: 
+      `edit 1 r=ChangeRemark`, their remark will be changed to ChangeRemark.
```

Comment:
slight nitpick, maybe use `r=Change remark` instead of `r=ChangeRemark` to show that spaces are supported? Same for `r=NewRemark` afterwards.